R/lightgbm.R

```diff
@@ -135,6 +135,7 @@ train_lightgbm <- function(x,
                            feature_pre_filter = FALSE,
                            free_raw_data = TRUE,
                            verbose = 0,
+                           save_tree_error = FALSE,
```

Comment:
I don't love that this argument doesn't match [the `lgbm.train` interface](https://lightgbm.readthedocs.io/en/stable/R/reference/lgb.train.html); previously I had tried adding a `valids` keyword argument for better alignment, but I thought that was too confusing alongside the existing `validation` keyword arg, and plus requiring the caller to format the valids set for record evals leaves more room for error in creating the `Dataset` object. This approach seemed like a reasonable middle ground but I'm open to alternatives!

Reply:
I like it! It's our shim package specifically for our pipeline. I think it's fine to add what we need as long as _we_ understand the interface.